lib/pbench/server/api/__init__.py

```diff
@@ -63,44 +57,35 @@ def create_app():
         sys.exit(1)
 
     try:
-        config = PbenchServerConfig(cfg_name)
+        server_config = PbenchServerConfig(cfg_name)
     except BadConfig as e:
         print(f"{__name__}: {e} (config file {cfg_name})", file=sys.stderr)
         sys.exit(1)
 
-    app = Flask(__name__)
-    api = Api(app)
+    app.config["SERVER_CONFIG"] = server_config.server_config
```

Comment:
Why are we going through the trouble of still using `app.config`?  Why not just reference the `server_config` instead?

Reply:
I was following the Flask config handling https://flask.palletsprojects.com/en/1.1.x/config/ where it says 

> Independent of how you load your config, there is a config object available which holds the loaded configuration values: The config attribute of the Flask object. This is the place where Flask itself puts certain configuration values and also where extensions can put their configuration values. But this is also where you can have your own configuration.

Flask `Config` is a subclass of a dictionary which comes with some pre-configured values and if we use extensions such as for example `flask_bcrypt` then we would need a `SECRET_KEY` or some key to bcrypt the password hash, this key can be stored as a config `key` (now it can also be just retrieved with PbenchServerConfig class, however some extensions can read the default value if the key is set). The advantage I see is that we dont have to pass the entire server_config object to other modules to get certain values (we just pass the app instance) and also for the testing when we import the `create_app` to make the `app_client` we can assign the `app_client.config = app.config` and therefore we so we dont need to make another serverconfig object

Reply:
but on the second thought I agree it adds an extra redundancy for our use case